src/python/antares_xpansion/config_loader.py

```diff
@@ -508,44 +509,47 @@ def _set_last_simulation_name(self):
         """
         return last simulation name
         """
-        last_dir_path = self.get_last_modified_dir(self.antares_output())
+        self._last_study = self.last_modified_study(self.antares_output())
 
-        if self.step() == "resume":
-            self._simulation_name = last_dir_path
+        if self.step() in ["resume", "sensitivity"] :
+            
+            self._xpansion_simulation_name = self._last_study
+            if self.is_zip(self._last_study):
+                self._xpansion_simulation_name = self._last_study.parent / self._last_study.stem
+                with zipfile.ZipFile(self._last_study, 'r') as output_zip:
+                    output_zip.extractall(self._xpansion_simulation_name)
```

Comment:
Lines 514 - 523 seem to be a duplicate of lines 91 - 99, is it possible to factorize ?